src/RoslynPad/MainWindow.xaml.cs

```diff
@@ -161,6 +165,55 @@ private void LoadWindowLayout()
         {
             FontSize = _viewModel.Settings.WindowFontSize.Value;
         }
+
+        ShiftWindowOntoScreen();
+    }
+
+    private void GetScreenAndWindowInfo()
+    {
+        var sb = new StringBuilder();
+
+        sb.AppendLine(_viewModel.Settings.WindowBounds);
+
+        var virtualScreenInfo = $"Virtual Screen: {SystemParameters.VirtualScreenLeft}, {SystemParameters.VirtualScreenTop}, {SystemParameters.VirtualScreenWidth}, {SystemParameters.VirtualScreenHeight}";
+
+        sb.AppendLine(virtualScreenInfo);
+
+        sb.AppendLine();
+        foreach (var s in System.Windows.Forms.Screen.AllScreens)
+        {
+            sb.AppendLine("Device Name: \t" + s.DeviceName);
+            sb.AppendLine("BitsPerPixel: \t" + s.BitsPerPixel);
+            sb.AppendLine("Bounds: \t\t" + s.Bounds);
+            sb.AppendLine("Primary: \t\t" + s.Primary);
+            sb.AppendLine("Working Area: \t" + s.WorkingArea);
+            sb.AppendLine("============================================================");
+        }
+
+        File.WriteAllText("temp.txt", sb.ToString());
+    }
+
+    private void ShiftWindowOntoScreen()
```

Comment:
this does work of shifting screen back into visible range on startup if it was closed on monitor that no longer exists.

Reply:
Instead of a separate method, I suggest you use `Math.Min/Max` in the bounds assignment above

Reply:
yeah, that was there mostly if you wanted to play with it. Removed.